specification/immunisation-fhir-api.yaml

```diff
@@ -2,40 +2,287 @@
+              example:
+                $ref: components/OperationOutcome/200_get-success.json
+          headers:
+            X-Correlation-ID:
+              $ref: '#/components/headers/Correlation-Id'
+            X-Request-ID:
+              $ref: '#/components/headers/Request-Id'
+        4XX:
+          $ref: '#/components/responses/4XX-imms'
+    delete:
+      summary: Delete an Immunisation record event by ID
+      operationId: deleteImmunisationRecordById
+      description: Logical deletion of an immunisation record event specified by ID. A delete interaction means that the resource can no longer found through a search interaction.
+      parameters:
+        - $ref: '#/components/parameters/ImmunisationEventID'
+        - $ref: '#/components/parameters/CorrelationID'
+        - $ref: '#/components/parameters/RequestID'
+      responses:
+        200:
+          description: Ok
+          content:
+            application/fhir+json:
+              schema:
+                $ref: '#/components/schemas/OperationOutcome'
+              example:
+                $ref: components/OperationOutcome/200_get-success.json
+          headers:
+            X-Correlation-ID:
+              $ref: '#/components/headers/Correlation-Id'
+            X-Request-ID:
+              $ref: '#/components/headers/Request-Id'
+        4XX:
+          $ref: '#/components/responses/4XX-imms'
+    put:
+      summary: Update an Immunisation record event by ID
+      operationId: updateImmunisationRecordById
+      description: The update interaction creates a new current version for an existing resource or creates an initial version if no resource already exists for the given id.
+      parameters:
+        - $ref: '#/components/parameters/ImmunisationEventID'
+        - $ref: '#/components/parameters/CorrelationID'
+        - $ref: '#/components/parameters/RequestID'
+      responses:
+        200:
+          description: Ok
+          content:
+            application/fhir+json:
+              schema:
+                $ref: '#/components/schemas/OperationOutcome'
+              example:
+                $ref: components/OperationOutcome/200_get-success.json
+          headers:
+            X-Correlation-ID:
+              $ref: '#/components/headers/Correlation-Id'
+            X-Request-ID:
+              $ref: '#/components/headers/Request-Id'
+        201:
+          description: Created
+          content:
+            application/fhir+json:
+              schema:
+                $ref: '#/components/schemas/OperationOutcome'
+              example:
+                $ref: components/OperationOutcome/201_create-success.json
+          headers:
+            X-Correlation-ID:
+              $ref: '#/components/headers/Correlation-Id'
+            X-Request-ID:
+              $ref: '#/components/headers/Request-Id'
+        4XX:
+          $ref: '#/components/responses/4XX-imms'
+
+components:
+
+  responses:
+    4XX-imms:
```

Comment:
It's good that we have a separate component just for 4XX. Can we separate them here? Like 401 with this specific operation outcome example. If it's possible just do it for only one. Like 400 INVALID is a good one because we have example. 
I'm just curious if it's possible to do it like that. i.e. referencing it in each endpoint as a single $ref and then in the definition of that component we expand it into each individual kind of error.